### PR TITLE
Introduce Typescript to griff-react

### PR DESCRIPTION
**Pull request by evancharlton**
Begin the process of migrating griff-react to Typescript. This initial
foray does the following:

  1. Introduces Typescript dependencies as dev dependencies.
  2. Relies on gts for the lint configuration.
  3. Builds to the build/ directory.
  4. Updates the stories and demo to point to that directory.
     (In the future, we should consider using NODE_PATH to do this
     automatically.)
  5. Migrates some of the items in src/utils/ to Typescript.

### CODE REVIEW

**Comment by cognite-cicd** [comment]
[pr-server]
The storybook for this PR is hosted on https://griff-react-pr-263.surge.sh

**Comment by cognite-cicd** [comment]
[pr-server]
The storybook for this PR is hosted on https://griff-react-pr-263.surge.sh

File: package.json [hunk 1]
```diff
@@ -23,7 +23,13 @@
...
     "lint": "eslint src stories",
     "prepush": "yarn lint",
-    "release": "rm -f .babelrc && yarn build && node scripts/publish-release.js"
+    "release": "rm -f .babelrc && yarn build && node scripts/publish-release.js",
+    "check": "gts check",
+    "compile": "tsc -p .",
+    "fix": "gts fix",
+    "prepare": "npm run compile",
+    "pretest": "npm run compile",
+    "posttest": "npm run check"
```
Comment (valerii-cognite) [inline package.json]: why not yarn?

File: package.json [hunk 2]
```diff
@@ -42,7 +48,13 @@
     "@storybook/addon-actions": "^4.1.11",
     "@storybook/addon-info": "^4.1.11",
     "@storybook/react": "^4.1.11",
+    "@types/d3": "^5.7.1",
+    "@types/react": "^16.8.8",
+    "@types/react-dom": "^16.8.3",
+    "awesome-typescript-loader": "^5.2.1",
     "axios": "^0.17.1",
+    "csstype": "^2.6.3",
+    "gts": "^0.9.0",
```
Comment (valerii-cognite) [inline package.json]: ok, let's give it a try

File: package.json (hunk 1 above)
Comment (valerii-cognite) [inline package.json]: `npm run check` throw me bunch of errors.

Comment (evancharlton) [reply package.json]: This was from gts -- I'll clean this up.

Comment (evancharlton) [reply package.json]: Yeah I don't know what this is all about. It's stuff like obscure dependencies not being installed, but shouldn't yarn take care of that?
```
error "jsdom#request#har-validator" not installed
error "@cognite/react-scripts#uglifyjs-webpack-plugin#find-up#locate-path" not installed
error "jsdom#request#mime-types" not installed
error "jsdom#request#oauth-sign" not installed
error "h2x-parse#jsdom#request" not installed
error "jsdom#request-promise-native#tough-cookie" not installed
error "jsdom#request#tough-cookie" not installed
error "jsdom#request-promise-native#request-promise-core" not installed
error "h2x-parse#jsdom#request-promise-native" not installed
error "jsdom#har-validator#ajv" not installed
error "h2x-parse#request#aws4" not installed
error "h2x-parse#request#combined-stream" not installed
```

(random snippet)

**Comment by valerii-cognite** [comment]
Couldbe because the command is running through npm but not yarn

**Comment by cognite-cicd** [comment]
[pr-server]
The storybook for this PR is hosted on https://griff-react-pr-263.surge.sh

File: package.json (hunk 1 above)
Comment (evancharlton) [reply package.json]: Yep, that's exactly it. It looks like the upcoming version of gts supports yarn -- I'll probably remove this until it's fully supported.

**Comment by cognite-cicd** [comment]
[pr-server]
The storybook for this PR is hosted on https://griff-react-pr-263.surge.sh

**Pull request merged**